packages/samples/headless-commerce-ssr-remix/app/routes/listings.$listingId.tsx

```diff
@@ -0,0 +1,78 @@
+import externalCartAPI from '@/client/external-cart-api';
+import externalContextAPI from '@/client/external-context-api';
+import {
+  listingEngineDefinition,
+  ListingStaticState,
+} from '@/lib/commerce-engine';
+import {getNavigatorContext} from '@/lib/navigator-context';
+import {
+  toCoveoCartItems,
+  toCoveoCurrency,
+} from '@/utils/external-api-conversions';
+import {NavigatorContext} from '@coveo/headless-react/ssr-commerce';
+import {LoaderFunctionArgs} from '@remix-run/node';
+import {useLoaderData, useParams} from '@remix-run/react';
+import invariant from 'tiny-invariant';
+import ProductList from '../components/product-list';
+import ListingProvider from '../components/providers/listing-provider';
+import {coveo_visitorId} from '../cookies.server';
+
+export const loader = async ({params, request}: LoaderFunctionArgs) => {
+  invariant(params.listingId, 'Missing listingId parameter');
```

Comment:
Wondering if bueno could do it?

Reply:
Probably, but do we want to show that developers should use Bueno for such things?

Reply:
I would not show bueno here. zod would be better as an example as it is the most popular one but I think this invariant dep is fine here